services/wfm/src/main/java/org/openkilda/wfm/topology/cache/FlowThrottlingBolt.java

```diff
@@ -0,0 +1,86 @@
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.wfm.topology.cache;
+
+import org.openkilda.messaging.Destination;
+import org.openkilda.messaging.Utils;
+import org.openkilda.messaging.command.CommandMessage;
+import org.openkilda.messaging.command.flow.FlowRerouteRequest;
+import org.openkilda.wfm.topology.AbstractTopology;
+import org.openkilda.wfm.topology.cache.service.ReroutesThrottling;
+import org.openkilda.wfm.topology.utils.AbstractTickRichBolt;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import org.apache.storm.task.OutputCollector;
+import org.apache.storm.task.TopologyContext;
+import org.apache.storm.topology.OutputFieldsDeclarer;
+import org.apache.storm.tuple.Tuple;
+import org.apache.storm.tuple.Values;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+
+public class FlowThrottlingBolt extends AbstractTickRichBolt {
+
+    private static final Logger logger = LoggerFactory.getLogger(FlowThrottlingBolt.class);
+
+    private final long timeout;
+
+    private final long maxTimeWindowSize;
+
+    private ReroutesThrottling reroutesThrottling;
+
+    public FlowThrottlingBolt(long timeout, long maxTimeWindowSize) {
+        this.timeout = timeout;
+        this.maxTimeWindowSize = maxTimeWindowSize;
+    }
+
+    @Override
+    public void prepare(Map conf, TopologyContext context, OutputCollector collector) {
+        super.prepare(conf, context, collector);
+        reroutesThrottling = new ReroutesThrottling(timeout, maxTimeWindowSize);
+    }
+
+    @Override
+    protected void doTick(Tuple tuple) {
+        for (Map.Entry<String, String> entry: reroutesThrottling.getReroutes().entrySet()) {
+            String flowId = entry.getKey();
+            String correlationId = entry.getValue();
+            FlowRerouteRequest request = new FlowRerouteRequest(flowId);
+            try {
+                String json = Utils.MAPPER.writeValueAsString(new CommandMessage(
+                        request, System.currentTimeMillis(), correlationId, Destination.WFM));
+                Values values = new Values(json);
+                _collector.emit(tuple, values);
+            } catch (JsonProcessingException exception) {
+                logger.error("Could not format flow reroute request by flow={}", flowId, exception);
+            }
+        }
+        _collector.ack(tuple);
+    }
+
+    @Override
+    protected void doWork(Tuple tuple) {
+        reroutesThrottling.putRequest(tuple.getString(0), tuple.getString(1));
```

Comment:
It would be more obviously if you load value from the tuple by name instead of depending on the position in the tuple.

Reply:
Fixed.